docs/content/preview/tutorials/AI/ai-llamaindex-openai.md

```diff
@@ -0,0 +1,250 @@
+---
+title: How to Develop RAG Apps with LlamaIndex, OpenAI and YugabyteDB
+headerTitle: Build RAG applications with LlamaIndex, OpenAI, and YugaybteDB
+linkTitle: LlamaIndex and OpenAI
+description: Learn to build RAG applications using LlamaIndex and OpenAI.
+image: /images/tutorials/ai/icons/llamaindex-icon.svg
+headcontent: Use YugaybteDB as the database backend for RAG applications
+menu:
+  preview:
+    identifier: tutorials-ai-llamaindex-openai
+    parent: tutorials-ai
+    weight: 60
+type: docs
+---
+
+This tutorial demonstrates how use LlamaIndex to build RAG (Retrieval-Augmented Generation) applications. By using the LlamaIndex [SQLJoinQueryEngine](https://docs.llamaindex.ai/en/stable/examples/query_engine/SQLJoinQueryEngine.html), the application can query a [PostgreSQL-compatible](https://www.yugabyte.com/postgresql/postgresql-compatibility/) YugabyteDB database from natural language. It can then infer whether to query a secondary vector index to fetch documents. In this case, the secondary index contains the Wikipedia pages of S&P 500 companies.
+
+## Prerequisites
+* Install Python3
+* Install Docker
+
+## Set up the application
+Download the application and provide settings specific to your deployment:
+
+1. Clone the repository.
+
+    ```sh
+    git clone https://github.com/YugabyteDB-Samples/yugabytedb-llamaindex-sp500-search.git
+    ```
+
+2. Install the application dependencies.
+
+    Dependencies can be installed in a virtual environment, or globally on your machine.
+
+    * Option 1 (recommended): Install Dependencies from *requirements.txt* in virtual environment
+        ```sh
+        python3 -m venv yb-llamaindex-env
+        source yb-llamaindex-env/bin/activate
+        pip install -r requirements.txt
+        # NOTE: Users with M1 Mac machines should use requirements-m1.txt instead:
+        # pip install -r requirements-m1.txt
+        ```
+
+    * Option 2: Install Dependencies Globally
+        ```sh
+        pip install llama-index
+        pip install psycopg2
+        # NOTE: Users with M1 Mac machines should install the psycopg2 binary instead:
+        # pip install psycopg2-binary
+        pip install python-dotenv
+        ```
+3. Create an [OpenAI API Key](https://platform.openai.com/api-keys) and store it's value in a secure location. This will be used to connect the application to the LLM to generate SQL queries, infer results and generate the proper response.
```

Comment:
I would delete this "and store it's value in a secure location". We assume that people know how to deal with their secrets.